account_payment_term_partner_holiday/models/account_invoice.py

```diff
@@ -17,7 +17,7 @@ def _onchange_payment_term_date_invoice(self):
 
     @api.onchange("date_due")
     def _onchange_date_due_account_payment_term_partner_holiday(self):
-        if self.date_due and self.payment_term_id and self.partner_id:
+        if self.date_due and self.partner_id:
```

Comment:
Also, shouldn't this be applied only when there is a payment term selected? Maybe the other fix should be enough and this should be restored...

Reply:
Due dates can be set without payment term, so if this condition is limiting the auto-change of the due date, then it should be removed.

Reply:
I have removed `payment_term_id` condition because `payment_term_id` field is not required to recalculate "allowed" `due_date`.

Reply:
Ok, it is just that the module name and all the docs suggest that this will only be applied when a payment term is selected (which makes sense as that is when the due date will be computed, in other cases it is manually selected, right?) Maybe we could update the docs to reflect that...